app.py

```diff
@@ -70,6 +71,44 @@
 
 app = Flask(__name__, static_url_path='')
 
+def into_number (string):
+    total = 0
+    for char in string:
+        total += ord (char)
+    return total
+
+def redirect_ab (request, session):
+    # If the user is logged in, we use their username as identifier, otherwise we use the session id
+    user_identifier = current_user(request) ['username'] or str (session_id)
```

Comment:
After thinking through this some more: session_id is a bit of a dangerous hashing key as well.

If the kid closes and reopens their browser, this may move them from inside the treatment group to outside the treatment group.

It may be disorienting if the UI of Hedy keeps changing.

Falling back to IP address instead might be safer...?